### PR TITLE
RDSK-9311: (gripper/input-controller) Guard against nil responses in rdk client/servers

### PR DESCRIPTION
**Pull request by martha-johnston**
went through encoder, gripper, input controller, motor, and posetracker server code to see what had the possibility to be returned as nil, and added checks to return errors in that case. also added tests for each nil failure. there were no changes for encoder, motor, or pose tracker.

added necessary tests to client test file to ensure that the server changes resulted in the same returned error, so the checks didn't need to be added in the client files as well.

still need to do this for the other components, but want to break up the work into smaller PRs (https://github.com/viamrobotics/rdk/pull/4618 & https://github.com/viamrobotics/rdk/pull/4621)

### CODE REVIEW

File: components/gripper/client_test.go [hunk 1]
```diff
@@ -145,6 +148,9 @@ func TestClient(t *testing.T) {
 		test.That(t, err, test.ShouldNotBeNil)
 		test.That(t, err.Error(), test.ShouldContainSubstring, errStopUnimplemented.Error())
 
+		_, err = client2.Geometries(context.Background(), extra)
+		test.That(t, err.Error(), test.ShouldContainSubstring, gripper.ErrGeometriesNil(failGripperName).Error())
```
Comment (oliviamiller) [inline components/gripper/client_test.go:152]: would` Errors.Is ` work here? I think we are trying to avoid using substring for errors test when possible

File: components/input/client_test.go [hunk 2]
```diff
@@ -252,11 +252,11 @@ func TestClient(t *testing.T) {
 
 		_, err = client2.Controls(context.Background(), map[string]interface{}{})
 		test.That(t, err, test.ShouldNotBeNil)
-		test.That(t, err.Error(), test.ShouldContainSubstring, errControlsFailed.Error())
+		test.That(t, err.Error(), test.ShouldContainSubstring, input.ErrControlsNil(failInputControllerName).Error())
 
 		_, err = client2.Events(context.Background(), map[string]interface{}{})
 		test.That(t, err, test.ShouldNotBeNil)
-		test.That(t, err.Error(), test.ShouldContainSubstring, errEventsFailed.Error())
+		test.That(t, err.Error(), test.ShouldContainSubstring, input.ErrEventsNil(failInputControllerName).Error())
```
Comment (oliviamiller) [inline components/input/client_test.go:259]: Should we still be testing the previous errors or is it not needed?

File: components/gripper/client_test.go (hunk 1 above)
Comment (martha-johnston) [reply components/gripper/client_test.go:152]: the problem is that it adds `"rpc error: code = Unknown desc = "` before the actual error, so figured it was easier to use substring. in the server code it doesn't add that string,  so you can just use `test.ShouldBeError`

Comment (martha-johnston) [reply components/gripper/client_test.go:152]: although I guess I didn't try `Errors.Is` perhaps that's easier, I've never seen that in the `test.That()` format, I'll try it out

File: components/input/client_test.go (hunk 2 above)
Comment (martha-johnston) [reply components/input/client_test.go:259]: it's not needed because `errControlsFailed` is just an error that was defined in this package and was only returned from the injected input controller. now we are testing if the real function returns the expected error
